### PR TITLE
[1369] Clean up feature specs with page objects

### PR DESCRIPTION
**Pull request by benilovj**
### Context
#228 was merged without using page objects in any of the changed code. This PR is cleaning this up.

### Changes proposed in this pull request
- add some extra structure into the templates to make testing easier/more robust
- push selectors out of the feature specs and into page objects

### CODE REVIEW

File: app/views/courses/index.html.erb [hunk 1]
```diff
@@ -30,15 +30,21 @@
   <p class="govuk-body-s govuk-!-margin-bottom-6">You’ll be taken to a Google Form to fill in your course details.</p>
 <% end %>
 
-<%= render partial: 'course_table', locals: { courses: @self_accredited_courses } if @self_accredited_courses %>
+<% if @self_accredited_courses %>
+  <section data-qa="courses__courses-table-section">
```
Comment (dankmitchell) [inline app/views/courses/index.html.erb]: Don't think we need to repeat `courses` here. `courses__table-section` is good enough

File: app/views/courses/index.html.erb [hunk 2]
```diff
@@ -30,15 +30,21 @@
...
 
 <% @courses_by_accrediting_provider.each do |accrediting_provider, courses| %>
-  <h2 class="govuk-heading-m">
-    <span class="govuk-caption-m">Accredited body</span>
-    <%= accrediting_provider %>
-  </h2>
-
-  <%= render partial: 'course_table', locals: { courses: courses } %>
+  <section data-qa="courses__courses-table-section">
+    <h2 class="govuk-heading-m">
```
Comment (dankmitchell) [inline app/views/courses/index.html.erb]: No need to add an extra level of HTML. I'd just add the data attribute on the `<h2>`

Comment (dankmitchell) [reply app/views/courses/index.html.erb]: NVM, I see the value of this now 👌

File: app/views/courses/index.html.erb (hunk 1 above)
Comment (benilovj) [reply app/views/courses/index.html.erb]: Fixed in 6e36e030540e6540f009da0072eba6453f02d2ec

**Pull request merged**